File: examples/go/vulnerable_app.go
package main

import (
	"crypto/md5"
	"database/sql"
	"fmt"
	"math/rand"
	"net/http"
	"os/exec"
)

func handler(db *sql.DB, w http.ResponseWriter, r *http.Request) {
	user := r.URL.Query().Get("user")
	targetURL := r.URL.Query().Get("url")
	command := r.URL.Query().Get("cmd")

	query := fmt.Sprintf("SELECT * FROM users WHERE name = '%s'", user)
	db.Query(query)

	http.Get(targetURL)
	exec.Command("sh", "-c", command).Output()

	token := fmt.Sprintf("%d", rand.Int())
	hash := md5.Sum([]byte(user + token))

	w.Write([]byte("<div>" + user + "</div>"))
	fmt.Println(hash)
}

File: examples/php/vulnerable_app.php
<?php

$name = $_GET["name"];
$url = $_GET["url"];
$cmd = $_GET["cmd"];
$payload = $_POST["payload"];

$query = "SELECT * FROM users WHERE name = '" . $name . "'";
$db->query($query);

echo "<div>" . $name . "</div>";
system($_GET["cmd"]);
file_get_contents($_GET["url"]);
unserialize($_POST["payload"]);

$token = md5($name . rand());
$password = "demo-insecure-password";

echo $token;
